api.go

```diff
@@ -64,6 +64,9 @@ func CreatePod(podConfig PodConfig) (*Pod, error) {
 		return nil, err
 	}
 
+	// Unlock the thread here after the network hook was run.
+	runtime.UnlockOSThread()
```

Comment:
It's a bit awkward to call that from that from every other entry points. Can't we try to call it when we know we're finished with the namespace setup?